php_phongo.c

```diff
@@ -962,12 +960,21 @@ bool php_phongo_read_preference_tags_are_valid(const bson_t *tags) /* {{{ */
 void php_phongo_read_preference_to_zval(zval *retval, const mongoc_read_prefs_t *read_prefs) /* {{{ */
 {
 	const bson_t *tags = mongoc_read_prefs_get_tags(read_prefs);
+	mongoc_read_mode_t mode = mongoc_read_prefs_get_mode(read_prefs);
 
 	array_init_size(retval, 2);
 
-	ADD_ASSOC_LONG_EX(retval, "mode", mongoc_read_prefs_get_mode(read_prefs));
+	switch (mode) {
+		case MONGOC_READ_PRIMARY: ADD_ASSOC_STRING(retval, "mode", "primary"); break;
+		case MONGOC_READ_PRIMARY_PREFERRED: ADD_ASSOC_STRING(retval, "mode", "primaryPreferred"); break;
+		case MONGOC_READ_SECONDARY: ADD_ASSOC_STRING(retval, "mode", "secondary"); break;
+		case MONGOC_READ_SECONDARY_PREFERRED: ADD_ASSOC_STRING(retval, "mode", "secondaryPreferred"); break;
+		case MONGOC_READ_NEAREST: ADD_ASSOC_STRING(retval, "mode", "nearest"); break;
```

Comment:
`add_assoc_string_ex()` requires a `char *` for the value string, so each of these will need `(char *)` casting. Alternatively, I suppose we can just add that in the macro definition.